samples/dnn/person_reid.cpp

```diff
@@ -0,0 +1,255 @@
+//  limitations under the License.
+// 
+//  Author: Xing Sun <winfredsun@tencent.com>
+//          Feng Zheng <zfeng02@gmail.com>
+//          Xinyang Jiang <sevjiang@tencent.com>
+//          Fufu Yu <fufuyu@tencent.com>
+//          Enwei Zhang <miyozhang@tencent.com>
+
+/*
+You can download the converted pb model from https://https://www.dropbox.com/s/7831tlmz8lajhpy/youtu_reid_baseline_lite.onnx
+or use your own onnx model yourself.
+*/
+#include <iostream>
+#include <fstream>
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/dnn.hpp>
+
+using namespace cv;
+using namespace cv::dnn;
+
+const char* keys =
+"{help    h  |                 | show help message}"
+"{model   m  |                 | network model}"
+"{query_list q |               | list of query images}"
+"{gallery_list g |             | list of gallery images}"
+"{batch_size | 32              | batch size of each inference}"
+"{resize_h   | 256             | height of input image}"
+"{resize_w   | 128             | width of input image}"
+"{topk k     | 5               | number of gallery images showed in visualization}"
+"{output_dir |                 | path for visualization(it should be existed)}"
+"{backend b  | 0               | Choose one of computation backends: "
+"0: automatically (by default), "
+"1: Halide language (http://halide-lang.org/), "
+"2: Intel's Deep Learning Inference Engine (https://software.intel.com/openvino-toolkit), "
+"3: OpenCV implementation }"
+"{target  t  | 0                | Choose one of target computation devices: "
+"0: CPU target (by default), "
+"1: OpenCL, "
+"2: OpenCL fp16 (half-float precision), "
+"3: VPU }";
+
+namespace cv{
+namespace reid{
+
+    static Mat preprocess(Mat img) 
+    {
+        const float mean[3] = {0.485, 0.456, 0.406};
+        const float std[3] = {0.229, 0.224, 0.225};
+        Mat ret = Mat(img.rows, img.cols, CV_32FC3);
+        for (int y = 0; y < ret.rows; y ++)
+        {
+            for (int x = 0; x < ret.cols; x++)
+            {
+                for (int c = 0; c < 3; c++)
+                {
+                    ret.at<Vec3f>(y,x)[c] = (img.at<Vec3b>(y,x)[c] / 255.0 - mean[2 - c]) / std[2 - c];
+                }
+            }
+        }
+        return ret;
+    }
+
+    static std::vector<float> normalization(std::vector<float> feature)
+    {
+        std::vector<float> ret;
+        float sum = 0.0;
+        for(int i = 0; i < (int)feature.size(); i++)
+        {
+            sum += feature[i] * feature[i];
+	}
+	sum = sqrt(sum);
+	for(int i = 0; i < (int)feature.size(); i++)
+	{
+	    ret.push_back(feature[i] / sum);
+	}
+	return ret;
+    }
+
+    static std::vector<std::vector<float>> extractFeatures(std::vector<std::string> imglist, Net* net, const int batch_size, const int resize_h, const int resize_w)
+    {
+	std::vector<std::vector<float>> features;
+	for(int st = 0; st < (int)imglist.size(); st += batch_size) {
```

Comment:
indentation is 4 spaces

https://github.com/opencv/opencv/wiki/Coding_Style_Guide#file-structure